youtube_dl/extractor/youmaker.py

```diff
@@ -0,0 +1,144 @@
+            "info_dict": {
+                "id": "71b5d2c5-31b6-43b8-8475-1dcb5e10dfb0",
+                "ext": "mp4",
+                "title": "Как сшить шапочку из трикотажа. Плоский шов двойной иглой.",
+                "description": r"re:(?s)^Привет друзья!\n\nВ этом видео я .* представлена www\.iksonmusic\.com$",
+                "thumbnail": r"re:^https?://.*\.(?:jpg|png)$",
+                "duration": 358,
+                "upload_date": "20190614",
+                "uploader": "user_318f21e00e1f8a6b414f20a654d0f4fc7d2053bc",
+                "timestamp": 1560554895,
+                "channel": "Sewing Ideas",
+                "channel_id": "40ca79f7-8b21-477f-adba-7d0f81e5b5fd",
+                "channel_url": r"re:https?://www.youmaker.com/channel/40ca79f7-8b21-477f-adba-7d0f81e5b5fd",
+                "tags": [
+                    "как сшить детскую шапочку из трикотажа",
+                    "как шить двойной иглой трикотаж",
+                ],
+            },
+            "params": {
+                "skip_download": True,
+            },
+        },
+    ]
+
+    def __init__(self, downloader=None):
+        """Constructor. Receives an optional downloader."""
+        super(YouMakerIE, self).__init__(downloader=downloader)
+        self.__protocol = "https"
+
+    def _fix_url(self, url):
+        if url.startswith("//"):
+            return "%s:%s" % (self.__protocol, url)
+        return url
+
+    @property
+    def _base_url(self):
+        return self._fix_url("//www.youmaker.com")
+
+    @property
+    def _api_url(self):
+        return "%s/v1/api" % self._base_url
+
+    @property
+    def _asset_url(self):
+        # as this url might change in the future
+        # it needs to be extracted from some js magic...
+        return self._fix_url("//vs.youmaker.com/assets")
+
+    def _extract_subtitles(self, system_id=None):
+        subtitles = {}
+
+        if system_id is None:
+            return subtitles
+
+        info = self._download_json(
+            "%s/video/subtitle?systemid=%s" % (self._api_url, system_id),
+            system_id,
+        )
+
+        if info.get("status") != "ok":
+            return subtitles
+
+        for item in info.get("data", []):
+            subtitles.setdefault(item["language_code"], []).append(
+                {"url": "%s/%s" % (self._asset_url, item["url"])}
+            )
+        return subtitles
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        if url.startswith("http://"):
+            self.__protocol = "http"
+        info = self._download_json(
+            "%s/video/metadata/%s" % (self._api_url, video_id), video_id
+        )
+
+        status = info.get("status", "something went wrong")
+        if status != "ok":
+            raise utils.ExtractorError(status, expected=True)
+
+        info = info["data"]
+        info["tags"] = [
+            tag.strip() for tag in info.get("tag", "").strip("[]").split(",") if tag
```

Comment:
`()` is a better 0-length iterable than `""`, which implies text.

Reply:
Yes, of course. Somehow I stopped reading the line too soon :-(

Reply:
But this is a sign, that I should strife for readability rather than conciseness.
I pushed some more commits, adding categories and hopefully making the code more readable and robust.
Can you please check?

Reply:
Actually I think the line I fingered was fine, but the replacement's fine too.